src/components/TimePicker.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import styled from 'styled-components'
+import { BaseDateTimePickerProps, DateTimePicker } from '@material-ui/pickers'
+import { TextField } from '@material-ui/core'
+import DateFnsAdapter from '@material-ui/pickers/adapter/date-fns'
+import { Control, UseFormMethods, FieldError, Controller } from 'react-hook-form'
+
+import { TradeFormData } from './TradeWidget'
+import { FormInputError } from './TradeWidget/FormMessage'
+
+interface DateTimePickerControlProps<T> extends BaseDateTimePickerProps {
+  control: Control<T>
+  onClose?: () => void
+  customOnChange: (date: Date, keyboardInputValue?: string | undefined) => void
+  formValues: {
+    value: string | null
+    setValue: UseFormMethods['setValue']
+    errors?: FieldError
+    inputName: keyof T
+  }
+}
+
+const DateTimePickerControl: React.FC<DateTimePickerControlProps<TradeFormData>> = ({
+  control,
+  formValues,
+  minDate = Date.now(),
+  customOnChange,
+  ...restProps
+}) => {
+  const memoizedDateAdapter = React.useMemo(() => {
+    return new DateFnsAdapter()
+  }, [])
+
+  const currentError = formValues.errors
+  return (
+    <Controller
+      control={control}
+      name={formValues.inputName}
+      render={(): JSX.Element => (
+        <DateTimePicker
+          {...restProps}
+          dateAdapter={memoizedDateAdapter}
+          value={formValues.value}
+          onChange={customOnChange}
+          inputFormat="yyyy/MM/dd HH:mm a"
+          ampm={false}
+          minDate={minDate}
+          renderInput={(props): JSX.Element => (
+            <TextField
+              {...props}
+              label="Set custom date"
+              name={formValues.inputName}
+              error={Boolean(currentError)}
```

Comment:
Not that it matters but everywhere else we do ``!!currentError`

Reply:
:+1: